hw_9.py

```diff
@@ -0,0 +1,302 @@
+# Задача-1
+# Реализовать дескриптор валидации для аттрибута email.
+# Ваш дескриптор должен проверять формат email который вы пытаетесь назначить
+import re
+
+class EmailDescriptor:
+
+    def __get__(self, instance, owner):
+        return self.email
+
+    def __set__(self, instance, value):
+        regex = '^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$' #
+        if re.search(regex, value):
+            print("Valid Email")
+            self.email = value
+        else:
+            raise AttributeError('Invalid Email')
+
+class MyClass:
+    email = EmailDescriptor()
+
+
+my_class = MyClass()
+my_class.email = "validemail@gmail.com"
+
+my_class.email = "novalidemail"
+# Raised Exception
+
+
+# Задача-2
+# Реализовать синглтон метакласс(класс для создания классов синглтонов).
+
+class Singleton(type):
+    cache = {}
+    def __call__(instance, *args, **kwargs):
+        if instance not in Singleton.cache:
+            Singleton.cache[instance] = super().__call__(*args, **kwargs)
+            return Singleton.cache[instance]
+        else:
+            return Singleton.cache[instance]
```

Comment:
else can be omitted

    def __call__(instance, *args, **kwargs):
        if instance not in Singleton.cache:
            Singleton.cache[instance] = super().__call__(*args, **kwargs)
        return Singleton.cache[instance]